src/Configuration/Profile.cs

```diff
@@ -287,31 +287,10 @@ public int
         public string WorldMapHiddenMarkerFiles { get; set; } = string.Empty;
 
 
-        internal static string ProfilePath { get; } = Path.Combine(CUOEnviroment.ExecutablePath, "Data", "Profiles");
-        internal static string DataPath { get; } = Path.Combine(CUOEnviroment.ExecutablePath, "Data");
-
         public static uint GumpsVersion { get; private set; }
 
-        public void Save(List<Gump> gumps = null)
+        public void Save(string path, List<Gump> gumps = null)
         {
-            if (string.IsNullOrEmpty(ServerName))
-            {
-                throw new InvalidDataException();
-            }
-
-            if (string.IsNullOrEmpty(Username))
-            {
-                throw new InvalidDataException();
-            }
-
-            if (string.IsNullOrEmpty(CharacterName))
-            {
-                throw new InvalidDataException();
-            }
-
-            string path = FileSystemHelper.CreateFolderIfNotExists
-                (ProfilePath, Username.Trim(), ServerName.Trim(), CharacterName.Trim());
-
```

Comment:
The path is now provided to this function.  All of the checks done here have been move to where the path string is created.